File: japan-go-fe/types/api/responses/kanji.response.ts
import { BaseResponse } from "./base.response";
import { VocabularyResponse } from "./vocabulary.response";

export interface KanjiResponse extends BaseResponse<number> {
    kanjiCharacter: string;
    unicode: string;
    kanjiVg: string;
    grade: number;
    strokeCount: number;
    frequency: number;
    jlptLevel: number;
    mainSinoVietnamese: string;
    onyomiList: string[];
    kunyomiList: string[];
    sinoVietnameseList: SinoVietnameseResponse[];
}

export interface KanjiPageResponse extends BaseResponse<number> {
    mainKanji: KanjiResponse;
    vocabularies: VocabularyResponse[];
}

export interface SinoVietnameseResponse extends BaseResponse<number> {
    readingText: string;
    sinoVietnameseMeaningList: string[];
}

File: japan-go-be/core/service/src/main/java/org/japan/dto/response/kanji/KanjiResponse.java
package org.japan.dto.response.kanji;

import lombok.*;
import lombok.experimental.FieldDefaults;
import lombok.experimental.SuperBuilder;
import org.japan.dto.response.base.BaseResponse;

import java.util.List;

@SuperBuilder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@FieldDefaults(level = AccessLevel.PRIVATE)
public class KanjiResponse extends BaseResponse {
    String kanjiCharacter;
    String unicode;
    String kanjiVg;
    Integer grade;
    Integer strokeCount;
    Integer frequency;
    Integer jlptLevel;
    String mainSinoVietnamese;
    List<String> onyomiList;
    List<String> kunyomiList;
    List<SinoVietnameseResponse> sinoVietnameseList;
}
